web-app/src/app/routes/monitor/monitor-form/monitor-form.component.html

```diff
@@ -107,6 +107,7 @@
             [extra]="{collectors}"
             [name]="'collector'"
             [(ngModel)]="collector"
+            (ngModelChange)="onCollectorChange($event)"
```

Comment:
There is no need to add a new event; if necessary, you can get the new value in the submit callback;

Reply:
https://github.com/apache/hertzbeat/blob/91111e55550d7d46a007db2f483b02a2dcd0a1a0/web-app/src/app/routes/monitor/monitor-new/monitor-new.component.ts#L168  
https://github.com/apache/hertzbeat/blob/91111e55550d7d46a007db2f483b02a2dcd0a1a0/web-app/src/app/routes/monitor/monitor-edit/monitor-edit.component.ts#L193  
Changing "`this.collector`" to "`info.collector`" in these two lines seems to be able to meet your requirement.

Reply:
Thanks for your suggestion. Please review again when you have time.